### PR TITLE
doc: add final-design.md doc after M2

### PR DESCRIPTION
**Pull request by Rqnsom**
Design document contains:
 - RPC descriptions
 - Extrinsics
 - Overall architecture

### CODE REVIEW

File: doc/initial_m1_design_document.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
+- [Introduction](#introduction)
+  - [Move language](#move-language)
+  - [Substrate framework](#substrate-framework)
```
[the remark this answers is not in the document]
Comment (Rqnsom) [reply doc/initial_m1_design_document.md]: Can be ignored - I kept the original name so we don't break any existing links to it.

**Review by neutrinoks — APPROVED** [review]
:+1:

File: doc/final-design.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
+# Introduction
+This article discusses the pallet-move functionality. 
+
+# Pallet API
+[smove] tool is used to build modules and bundles and to create script transactions (with given script arguments). The output from the `smove` tool in the Move projects is the main input for below extrinsics.
+
+MoveVM scripts and modules are not allowed to run forever - therefore, the `gas` value is used for that purpose. The provided `gas` gets converted to `Weight`, a mechanism that prevents extriniscs from running forever.
+The script execution is an atomic operation - if the provided gas is insufficient, the MoveVM within the pallet will reject the script and no changes shall occur.
```
Comment (asmie) [inline doc/final-design.md:8]:
```suggestion
The script execution is an atomic operation - if the provided gas is insufficient, the MoveVM within the pallet will reject the script, and no changes shall occur.
```

File: doc/final-design.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+# Introduction
+This article discusses the pallet-move functionality. 
+
+# Pallet API
+[smove] tool is used to build modules and bundles and to create script transactions (with given script arguments). The output from the `smove` tool in the Move projects is the main input for below extrinsics.
```
Comment (asmie) [inline doc/final-design.md:5]:
```suggestion
[smove] tool is used to build modules and bundles and to create script transactions (with given script arguments). The output from the `smove` tool in the Move projects is the main input for the below extrinsics.
```

**Review by asmie — APPROVED** [review]
There could be one more sentence about charging user full price for the script execution even if it fails.

:+1:

File: doc/final-design.md (hunk 3 above)
Comment (Rqnsom) [reply doc/final-design.md:5]: Done

File: doc/final-design.md (hunk 2 above)
Comment (Rqnsom) [reply doc/final-design.md:8]: Done

**Review by Rqnsom — COMMENTED** [review]
> There could be one more sentence about charging user full price for the script execution even if it fails.

Good idea. Done!

**Review by asmie — APPROVED** [review]

**Pull request merged**